nerfacc/estimators/occ_grid.py

```diff
@@ -271,6 +345,9 @@ def _sample_uniform_and_occupied_cells(self, n: int) -> List[Tensor]:
             uniform_indices = torch.randint(
                 self.cells_per_lvl, (n,), device=self.device
             )
+            # filter out the cells with -1 density (non-visible to any camera)
+            cell_ids = lvl * self.cells_per_lvl + uniform_indices
+            uniform_indices = uniform_indices[self.occs[cell_ids] >= 0.0]
```

Comment:
Yes, we should also include this in `_get_all_cells` since we update all cells during the warm-up stage.